backend/routes/documents.py

```
@@ -280,13 +310,180 @@ def update_document(document_id: str, request: Request, body: dict = Body(...)):
+    The request_id column may not exist in older schemas — catch that and
+    return None so deployments can ship the code before the migration runs.
+    Defensive against non-list return shapes (mocked DBs, partial errors)
+    so a misbehaving Supabase response can't masquerade as a cache hit.
+    """
+    try:
+        rows = table("documents").select(
+            "id,user_id,course_id,file_name,category,summary,concept_notes,created_at,processed_at",
+            filters={"user_id": f"eq.{user_id}", "request_id": f"eq.{request_id}"},
+            limit=1,
+        )
+    except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)